Python SDK fix: fasthtml correct imports and raise version (#576)

axelknock opened the pull request:
Corrects imports (they were being shadowed earlier so I got the wrong names) and raised version which should have been raised before.

axelknock commented:
Should also mention [#707](https://github.com/starfederation/datastar/issues/707#issuecomment-2683832434):
> Can you please bump the version, [@lllama](https://github.com/lllama), or remind me what I need to do if it requires some action from my side?
- @bencroker

I'll look at pypi package deployment for the python SDK.

bencroker commented:
Review please, @lllama.

dalito commented on sdk/python/pyproject.toml:
```diff
@@ -37,11 +37,12 @@ build-backend = "hatchling.build"
     "sanic>=24.6.0",
     "uvicorn>=0.32.1",
+    "bump-my-version>=0.32.0",
 ]
```
Above the version is declared as dynamic. But then you use hard-coded version below and add a tool here. So is the idea to use dynamic versioning or not?

axelknock replied:
I added this because @lllama included a declaration for the bumpmyversion tool. When I tried to use the tool it failed due to the repo tags interfering with the SDK's versioning (and I could not figure out how to disregard the tag version), so it might be worth discarding both this and the tool configuration, but as far as I can tell @lllama currently manually distributes the SDK to pypi so I defer to him.

lllama replied:
So I use `uv` for building and publishing the sdk to pypi. I think it'll probably be best to use `uv` for running `bump-my-version` as well.

axelknock replied:
@lllama I can't manage to run bump-my-version inside the datastar repo without the repo's tags interfering with the command. I don't see any obvious way to ignore the tags. How do you manage to run the command? Do you have the python SDK outside the repo?

dalito commented on sdk/python/pyproject.toml:
```diff
@@ -37,11 +37,12 @@ build-backend = "hatchling.build"
     "django>=4.2.17",
     "fastapi[standard]>=0.115.4",
-    "fasthtml>=0.12.0",
+    "python-fasthtml==0.12.0",
```
Is a hard pin for an SDK is a good idea? Is it really required for python-fasthtml?

axelknock replied:
True.

axelknock committed:
feat(python): Use dynamic minimum version for fasthtml

axelknock commented:
@bencroker @lllama said on discord that there are too many unrelated changes in this PR. I will cut it back tomorrow and comment when it's done.

axelknock committed:
fix(python): correct fasthtml imports for sse responses

axelknock commented:
Apologies for not getting this done sooner @bencroker, @lllama. I've cut everything but the fix and dynamic fasthtml version.

Kvit commented:
@lllama  all still works in alex's branch

bencroker commented:
Review please, @lllama.

axelknock committed:
feat(sdk): environment flag for fasthtml python version

Since fasthtml's minimum version is higher than datastar-py's, an environment flag lets package
managers only install it if the python version is high enough.

axelknock commented:
I made a change suggested by @gazpachoking on discord to put the fasthtml requirement behind an env variable since its minimum python version is higher than datastar-py. I confirmed it works with uv sync. Let me know if I should make any other changes @lllama

gazpachoking commented:
This is still valid isn't it @axelknock?

axelknock commented:
@gazpachoking I screwed up when making this PR by using my `develop` branch and I don't want it tied to this PR anymore. I'll open a new PR for the fix.
